src/WebForms/UI/Control.cs

```diff
@@ -370,6 +371,50 @@ public virtual string ClientID
     ]
     protected internal virtual HttpContext Context => HttpContext.Current;
 
+    protected virtual ControlAdapter ResolveAdapter()
+    {
+        if (flags[controlAdapterResolved])
+        {
+            return AdapterInternal;
+        }
+        if (DesignMode)
+        {
+            flags.Set(controlAdapterResolved);
+            return null;
+        }
+
+        HttpContext context = Context;
+        if (context != null && context.Request.Browser != null)
+        {
+            AdapterInternal = null;
+
+//TODO create issue
+#if   GetAdapter_BrowserCapabilities
```

Comment:
@twsouthwick should we handle here or in the adapter ?

Reply:
create an issue here and mark this with PORT_BROWSERCAPABILITIES  - I like the `PORT_` prefix for easily identifiying what's needed